crates/ruff/src/rules/pycodestyle/rules/logical_lines/mod.rs

```diff
@@ -88,22 +88,34 @@ impl<'a> LogicalLines<'a> {
 
         let mut builder = LogicalLinesBuilder::with_capacity(tokens.len());
         let mut parens: u32 = 0;
+        let mut non_empty = false;
 
         for (token, range) in tokens.iter().flatten() {
             let token_kind = TokenKind::from_token(token);
             builder.push_token(token_kind, *range);
 
             match token_kind {
+                TokenKind::Newline if parens == 0 => {
+                    builder.finish_line();
+                    non_empty = false;
+                }
+                TokenKind::NonLogicalNewline if parens == 0 => {
+                    if non_empty {
+                        builder.finish_line();
+                        non_empty = false;
+                    }
```

Comment:
```suggestion
                TokenKind::NonLogicalNewline if parens == 0 && non_empty => { 
              		builder.finish_line();
```

Reply:
Moved the checks into the builder instead, which is necessary to handle `builder.finish()` properly (which calls `.finish_line()` internally).